app-backend/datamodel/src/main/scala/com/azavea/rf/datamodel/package.scala

```diff
@@ -18,6 +18,19 @@ import geotrellis.proj4._
 import geotrellis.slick.Projected
 
 package object datamodel {
+
+  trait OwnerCheck {
+    def checkOwner(createUser: User, ownerUserId: Option[String]): String = {
+      (createUser, ownerUserId) match {
+        case (user, Some(id)) if user.id == id => user.id
+        case (user, Some(id)) if user.isInRootOrganization => id
+        case (user, Some(id)) if !user.isInRootOrganization =>
+          throw new IllegalArgumentException("Insufficient permissions to set owner on object")
```

Comment:
In testing I noticed that this returns a `400` rather than a `403` status code as I would expect. I know that our error handling is sometimes awkward with respect to status codes but if there's an easy fix it would be good to return a `403` here.